cmd/ebrelayer/relayer/cosmos.go

```diff
@@ -127,20 +139,44 @@ func getOracleClaimType(eventType string) types.Event {
 
 // Parses event data from the msg, event, builds a new ProphecyClaim, and relays it to Ethereum
 func (sub CosmosSub) handleBurnLockMsg(attributes []tmKv.Pair, claimType types.Event) error {
+
 	cosmosMsg, err := txs.BurnLockEventToCosmosMsg(claimType, attributes)
 	if err != nil {
 		fmt.Println(err)
 		return err
 	}
 	sub.Logger.Info(cosmosMsg.String())
 
-	// TODO: Ideally one validator should relay the prophecy and other validators make oracle claims upon that prophecy
+	// Only deal with submit message
+	if cosmosMsg.MessageType != ethbridge.MsgSubmit {
+		return nil
+	}
+
 	prophecyClaim := txs.CosmosMsgToProphecyClaim(cosmosMsg)
-	err = txs.RelayProphecyClaimToEthereum(sub.EthProvider, sub.RegistryContractAddress,
-		claimType, prophecyClaim, sub.PrivateKey)
+	cethAmount := cosmosMsg.CethAmount.BigInt()
+
+	gasUsed, err := txs.RelayProphecyClaimToEthereum(sub.EthProvider, sub.RegistryContractAddress,
+		claimType, prophecyClaim, sub.PrivateKey, cethAmount)
+
+	// If failed to send prophecy claim to Ethereum
 	if err != nil {
 		fmt.Println(err)
+
+		if err.Error() == "not enough ceth to cover the gas costs" {
```

Comment:
This should be a constant - it's used in multiple places and we care about exact string equality

Reply:
I will define a constant